src/x/resource/types.go

```diff
@@ -50,3 +50,16 @@ type CloserFn func()
 func (fn CloserFn) Close() {
 	fn()
 }
+
+// ErrCloser is an object that can be closed which returns an error.
+type ErrCloser interface {
```

Comment:
nit: What are your thoughts on using `Closer` to be synonymous with `io.Closer`, and `SimpleCloser` to be `func Close()`?

Reply:
So are you proposing 
- `ErrCloser` replaced by `io.Closer`
- `Closer` renamed `SimpleCloser` (i.e. the void version of `io.Closer`)

I suppose using `io.Closer` would be nice since it is native, while nit-downside is technically not all uses would actually be IO-related.

Reply:
Almost - I'm saying that:

```go
type Closer interface {
  Close() error // matches io.Closer
}

type SimpleCloser interface {
  Close() // no stdlib analogue
}
```

keeps things in line with how they're used elsewhere. Agree that "not everything is IO-based", but `io.Closer` is the de facto general purpose "closer type" - so, while I don't think we should omit `Closer` from this package (and therefore would need to change `resource.Closer` refs to `io.Closer), I do think that `resource.Closer` should match `io.Closer` for consistency.

(Also, the things we're closing do in fact tend to be at least quasi-IO-based in a lot of cases, because <resources>.)

Whether the type is redefined here or is simply aliased here, I don't have a strong preference necessarily:

```go
type Closer = io.Closer // ok

type Closer io.Closer // also ok

type Closer interface {
  Close() error
} // also ok
```

Functionally I don't think it matters a ton since it's unlikely that that API will change any time soon.

Reply:
+1 having `Closer` match the same signature as `io.Closer` by returning an error; I think folks would find this more familiar

Reply:
Ah got it. Yes matching the signature for `Closer` makes sense. Will do